subprojects/base-services/src/main/java/org/gradle/internal/jvm/Jvm.java

```diff
@@ -354,64 +340,29 @@ private File findToolsJar(File javaHome) {
         return null;
     }
 
-    public Map<String, ?> getInheritableEnvironmentVariables(Map<String, ?> envVars) {
-        return envVars;
-    }
-
-    public boolean isIbmJvm() {
-        return false;
-    }
-
-    /**
-     * Details about a known JVM implementation.
-     */
-    static class JvmImplementation extends Jvm {
-        JvmImplementation(OperatingSystem os) {
-            super(os);
-        }
-    }
-
-    static class IbmJvm extends JvmImplementation {
-        IbmJvm(OperatingSystem os) {
-            super(os);
-        }
-
-        @Override
-        public boolean isIbmJvm() {
-            return true;
+    public static Map<String, ?> getInheritableEnvironmentVariables(Map<String, ?> envVars) {
+        Map<String, Object> vars = new HashMap<String, Object>();
+        for (Map.Entry<String, ?> entry : envVars.entrySet()) {
+            // The following are known variables that can change between builds and should not be inherited
+            if (entry.getKey().matches("APP_NAME_\\d+")
+                || entry.getKey().matches("JAVA_MAIN_CLASS_\\d+")
+                || entry.getKey().equals("TERM_SESSION_ID")
+                || entry.getKey().equals("ITERM_SESSION_ID")) {
+                continue;
+            }
+            vars.put(entry.getKey(), entry.getValue());
         }
+        return vars;
     }
 
-    /**
-     * Note: Implementation assumes that an Apple JVM always comes with a JDK rather than a JRE, but this is likely an over-simplification.
-     */
-    static class AppleJvm extends JvmImplementation {
-        AppleJvm(OperatingSystem os) {
-            super(os);
-        }
-
-        /**
-         * {@inheritDoc}
-         */
-        @Override
-        public File getToolsJar() {
-            return null;
```

Comment:
Is the behavior change here for `getToolJar` for apple JVMs significant?

Reply:
I think this was a minor optimization because it was unlikely to exist on Apple JVMs.  I went ahead and made it so that once it looks for the tools jar once, it caches the "not found" value for subsequent calls.  This only gets called off of `Jvm.current()` in any production code, so this should only get calculated once (and then cached) which shouldn't have any noticeable impact.